File: Solutions/1481. Least Number of Unique Integers after K Removals/1481.py
class Solution:
    def findLeastNumOfUniqueInts(self, arr: List[int], k: int) -> int:
        fre = Counter(arr)
        heap = list(fre.values())
        heapq.heapify(heap)
        ans = len(heap)
        while k > 0 and heap:
            f = heapq.heappop(heap)
            if k >= f:
                k -= f
                ans -= 1
        return ans

File: Solutions/1481. Least Number of Unique Integers after K Removals/1481.java
class Solution {
    public int findLeastNumOfUniqueInts(int[] arr, int k) {
        Map<Integer, Integer> fre = new HashMap<>();
        for(int a : arr){
            fre.put(a, 1 + fre.getOrDefault(a,0));
        }
        PriorityQueue<Integer> pq = new PriorityQueue<>(fre.values());
        while (k > 0)
            k -= pq.poll();
        return k < 0 ? pq.size() + 1 : pq.size(); 
    }
}